packages/edit-site/src/components/sidebar-edit-mode/page-panels/reset-default-template.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * WordPress dependencies
+ */
+import { useDispatch } from '@wordpress/data';
+import { MenuGroup, MenuItem } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useEntityRecord } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import {
+	useCurrentTemplateSlug,
+	useEditedPostContext,
+	useIsPostsPage,
+} from './hooks';
+import { store as editSiteStore } from '../../../store';
+
+export default function ResetDefaultTemplate( { onClick } ) {
+	const currentTemplateSlug = useCurrentTemplateSlug();
+	const isPostsPage = useIsPostsPage();
+	const { postType, postId } = useEditedPostContext();
+	const entitiy = useEntityRecord( 'postType', postType, postId );
+	const { setPage } = useDispatch( editSiteStore );
+	const { createSuccessNotice } = useDispatch( noticesStore );
+	// The default template in a post is indicated by an empty string.
+	if ( ! currentTemplateSlug || isPostsPage ) {
+		return null;
+	}
+	return (
+		<MenuGroup>
+			<MenuItem
+				onClick={ async () => {
+					entitiy.edit( { template: '' }, { undoIgnore: true } );
+					await entitiy.save();
```

Comment:
Is the saving necessary?

Reply:
The problem with this code is that it will save all the other entity changes that were pending, not just the template.